determined-ai/determined
# fix: run scheduling on agent connection/enable events, reconnectBacklog replay.

## Description

We didn't run scheduler in case when an agent reconnects or gets enabled, and no other action is happening. This'd mean a pending task would stay pending, even though a new capacity has appeared.

- add an auto test for this.
- add a bit of a more structured logging for agent messages in RP actor.
- also remove unused `uuid` field on agent.
- also remove unused `internal/sproto/container.go`, gone as of #3663.

## Test Plan
In a single agent setup,
- Disable the agent.
- Submit a task.
- Enable the agent.
- It should get scheduled.

## Commentary (optional)




## Checklist

See [Release Note](../docs/release-notes/README.md) for details.
